### PR TITLE
Added informations about the MySQL grammar from Oracle

### PR DESCRIPTION
**Pull request by mike-lischke**
I cannot directly add the MySQL grammar from Oracle here, as it has some dependencies and needs to be viewed in context, but I added a folder with a readme file that contains the relevant information to get it.

Since there is already another grammar, I moved both into own subfolders, to avoid confusion.

### CODE REVIEW

**Comment by mike-lischke** [comment]
Can someone please advice what I have to do to make Travis happy? I'm not a Java developer and the set up there is completely unknown to me. I tried something obvious, but that failed, so I'm a bit helpless here now. Thanks.

**Comment by teverett** [comment]
It looks like the Java compilation is failing

[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project mysql: Compilation failure: Compilation failure: 
[ERROR] /home/travis/build/antlr/grammars-v4/mysql/target/generated-sources/antlr4/Positive Technologies/MySqlParser.java:[2,17] ';' expected
[ERROR] /home/travis/build/antlr/grammars-v4/mysql/target/generated-sources/antlr4/Positive Technologies/MySqlParserVisitor.java:[2,17] ';' expected
[ERROR] /home/travis/build/antlr/grammars-v4/mysql/target/generated-sources/antlr4/Positive Technologies/MySqlParserBaseVisitor.java:[2,17] ';' expected
[ERROR] /home/travis/build/antlr/grammars-v4/mysql/target/generated-sources/antlr4/Positive Technologies/MySqlParserListener.java:[2,17] ';' expected
[ERROR] /home/travis/build/antlr/grammars-v4/mysql/target/generated-sources/antlr4/Positive Technologies/MySqlParserBaseListener.java:[2,17] ';' expected
[ERROR] /home/travis/build/antlr/grammars-v4/mysql/target/generated-sources/antlr4/Positive Technologies/MySqlLexer.java:[2,17] ';' expected
[ERROR] -> [Help 1]

**Comment by mike-lischke** [comment]
I'm sorry, but that makes not much sense to me. Do all these files miss a semicolon at the same position? I generated the files locally and they seem to be fine. There must be something else, which is the true culprit (space in folder name?).

File: mysql/Oracle/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Oracle's MySQL grammar
+
+##General
+Since more than a decade the MySQL GUI dev tools team at Oracle offers the open source [MySQL Workbench](https://github.com/mysql/mysql-workbench) product, which uses ANTLR4 for all MySQL code parsing tasks. This requires to translate all changes from the [MySQL server grammar](https://github.com/mysql/mysql-server/blob/8.0/sql/sql_yacc.yy) to ANTLR4, which is an ongoing effort to always stay up-to-date with the lastest and greatest server features. The current grammar supports the server versions 5.6, 5.7 and 8.0.
```
Comment (KvanTTT) [inline mysql/Oracle/README.md]: There is a good practice of breaking long text lines into several short ones (up to 80-120 chars). It's better for comparison (diff) of different versions and more convenient for review to point certain line, not the whole paragraph.

**Comment by mike-lischke** [comment]
> * Maybe it makes sense to move your grammar to this repository if the first is more or less ready?
 
I wrote in the description that I cannot do that.

> * I prefer more simple directory names without spaces and hyphens ideally in lower case: `Positive-Technologies` -> `pt` or `PT` or just `community`.

I prefer names that a human can easily read and understand. This is a company name and I certainly don't want to obfuscate this, because some ancient systems cannot cope with space chars.

File: mysql/Oracle/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+The server version and the SQL mode can be switched at runtime, enabling the use of a single parser with different version/mode settings and to provide better error messages (like for [a feature that is only valid for a specific version](https://github.com/mysql/mysql-workbench/blob/8.0/modules/db.mysql.parser/src/mysql_parser_module.cpp#L391)).
+
+##Using the Grammar
+The [MySQL ANTLR4 grammar](https://github.com/mysql/mysql-workbench/tree/8.0/library/parsers/grammars) in MySQL Workbench considers these conditions to deliver exact parsing results. Unfortunately, this is not possible without using action code (and predicates) in the grammar, which are written in the target language of the host application (C++). Additionally, the generated parser + lexer classes use a common base class to provide support code and which contain fields for the current server version and active SQL modes (which are then used in the predicates to guide the parsing process). You can find the few extra files in [another subfolder of the parser library](https://github.com/mysql/mysql-workbench/tree/8.0/library/parsers/mysql).
```
[the remark this answers is not in the document]
Comment (mike-lischke) [reply mysql/Oracle/README.md]: You mean using functions + fields from a base class? That's exactly what I'm doing there.

File: mysql/Oracle/README.md (hunk 1 above)
Comment (mike-lischke) [reply mysql/Oracle/README.md]: I'm not sure I follow that argumentation about better comparison, but I added linebreaks anyway. They don't change how the file is rendered in HTML, but increase the plain text readability.

**Comment by KvanTTT** [comment]
Ok, thanks for a contribution!

**Comment by mike-lischke** [comment]
@KvanTTT btw. you can essentially close all the [MySQL issues](https://github.com/antlr/grammars-v4/issues?q=is%3Aissue+is%3Aopen+label%3Amysql) by referring to my grammar. It does all that correctly.

**Comment by KvanTTT** [comment]
Not sure it's correct, because they are related to previous universal grammar. At least users should be asked whether their issues have been resolved.

File: mysql/Oracle/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (KvanTTT) [reply mysql/Oracle/README.md]: Ok, if so I think it should be not very complicated to port this grammar to other targets.

**Pull request merged**